### PR TITLE
Streamline the pseudocode describing SignOp

### PR DESCRIPTION
**Pull request by burmako**
Inspired by #1382, this PR: 1) reduces nesting, 2) drops the final ifs from the is_integer and is_float code blocks, 3) makes it clear that the NaN case for complex numbers returns NaN for both real and imaginary components.

### CODE REVIEW

File: docs/spec.md
Comment (ghpvnist) [inline docs/spec.md:4947]: abs(x) returns a float, so how about explicitly writing:
```
return divide(x, (abs(x).real, abs(x).imag))
```

**Review by ghpvnist — APPROVED** [review]
One comment and lgtm.

File: docs/spec.md
Comment (burmako) [reply docs/spec.md:4947]: "abs(x) returns a float". Good point, we've also discussed this in #1382. You probably meant `divide(x, (abs(x), 0))`, right? Anyway, I used convert instead to hopefully make this more explicit.

"We should also have a separate case for handling zeros (regardless of sign should return positive zero)". Is this behavior specced somewhere? Here are the data points that seem relevant:
  * [elemental_ir_emitter.py](https://github.com/tensorflow/tensorflow/blob/f6ca332b93cc5584dd09ca991865a6a7091f3eff/tensorflow/compiler/xla/service/elemental_ir_emitter.cc#L1054-L1064) does what's proposed in the comment above.
  * [lax.py](https://github.com/google/jax/blob/d9383fc80dc739443fd10365c85c7e781e370b01/jax/_src/lax/lax.py#L234-L235) only says `x / |x|`.
  * Surprisingly, [numpy.sign](https://numpy.org/doc/stable/reference/generated/numpy.sign.html) does something quite different: `sign(x.real) + 0j if x.real != 0 else sign(x.imag) + 0j`, so we shouldn't probably index too much on that.

Comment (ghpvnist) [reply docs/spec.md]: Ah yes, I meant `(abs(x), 0)`. Handling zeros is a specific behavior in `elemental_ir_emitter.py`. And though the jax doc you linked seem to suggest otherwise, as mentioned in https://github.com/openxla/stablehlo/pull/1382#discussion_r1165960432, lowering to hlo in [lax.py](https://github.com/google/jax/blob/d9383fc80dc739443fd10365c85c7e781e370b01/jax/_src/lax/lax.py#L1714-L1724) where the `sign_p` comes from the [line](https://github.com/google/jax/blob/d9383fc80dc739443fd10365c85c7e781e370b01/jax/_src/lax/lax.py#L237) under the doc also follows this behavior.

File: docs/spec.md
Comment (burmako) [inline docs/spec.md:4946]: Based on the outcome of the conversation about how `(0.0, 0.0)` should be handled by SignOp, we will want to enable and/or change `stablehlo/testdata/sign_special_0_dtype_complex64.mlir`.

File: docs/spec.md
Comment (burmako) [reply docs/spec.md]: Upon further reflection, I am convinced now that it is a good idea to specialcase `(0.0, 0.0)` in the specification, consistently with how `0.0` is specialcased as well. I've updated the PR accordingly.

File: docs/spec.md
Comment (burmako) [reply docs/spec.md:4946]: Will wait for #1382 to be merged and will then update this PR to include interpreter changes to accommodate `(0.0, 0.0)` and enable this test.

**Comment by burmako** [comment]
Getting spurious CMake build errors across multiple pull requests:

```
Ign:1 http://azure.archive.ubuntu.com/ubuntu jammy/universe amd64 lld amd64 1:14.0-55~exp2
Ign:1 http://azure.archive.ubuntu.com/ubuntu jammy/universe amd64 lld amd64 1:14.0-55~exp2
Ign:1 http://azure.archive.ubuntu.com/ubuntu jammy/universe amd64 lld amd64 1:14.0-55~exp2
Err:1 http://azure.archive.ubuntu.com/ubuntu jammy/universe amd64 lld amd64 1:14.0-55~exp2
  Could not connect to azure.archive.ubuntu.com:80 (52.147.219.192), connection timed out
E: Failed to fetch [http://azure.archive.ubuntu.com/ubuntu/pool/universe/l/llvm-defaults/lld_14.0-55%7eexp2_amd64.deb](http://azure.archive.ubuntu.com/ubuntu/pool/universe/l/llvm-defaults/lld_14.0-55~exp2_amd64.deb)  Could not connect to azure.archive.ubuntu.com:80 (52.147.219.192), connection timed out
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

I've just run a clean CMake build locally, and it seems to work, so I think that this is safe to merge. I'll rerun the build again on my machine just to be sure and will force merge this PR once the local build succeeds.

**Pull request merged**
